ompi/mpi/c/type_get_contents.c

```diff
@@ -65,6 +66,13 @@ int MPI_Type_get_contents(MPI_Datatype mtype,
         }
     }
 
+    /* Counts may exceed actual ones, we have no choice but to recompute them */
+    rc = ompi_datatype_get_args(mtype, 0, &max_integers, NULL, &max_addresses, NULL, &max_datatypes,
+                                NULL, &type_code);
+    if (rc != MPI_SUCCESS) {
+        OMPI_ERRHANDLER_RETURN(MPI_ERR_INTERN, MPI_COMM_WORLD, MPI_ERR_INTERN, FUNC_NAME);
+    }
+
```

Comment:
What is the point of querying the sizes if we don't act on that information? If the user told us the arrays are of a certain size and the type requires more then we should error out, raising `MPI_ERR_ARG`.

Reply:
This updates the user provided max lengths before passing them to `ompi_datatype_get_args`. As if `ompi_datatype_get_args` was not correctly handling the case where user provides more space than required (which is not the case as I pointed in my prior comment).

Reply:
I'd be worried about the converse: the user provides insufficient space and we write past the bounds of the user buffer. That is what I thought this patch addressed. Otherwise I don't see what this patch accomplishes.